src/custom/state/multicall/utils.ts

```diff
@@ -0,0 +1,202 @@
+import { Contract } from '@ethersproject/contracts'
+
+import { Call, ListenerOptions } from 'state/multicall/actions'
+import { CallResult, isValidMethodArgs, OptionalMethodInputs, Result, toCallState } from 'state/multicall/hooks'
+import { Multicall2 } from 'abis/types'
+import { fetchChunk } from 'state/multicall/updater'
+
+export type CallParams = {
```

Comment:
All the code of this file, comes from the `hook` version of Uniswap? 

If this is the case, maybe is good to give a hint where do you get it from, and the motivation for the duplication, and what is changing. 

I find it a bit hard to revie, although I think the main reason is that the original code was hard to review too.  Any hints to make it easier?